code-name-taran/src/com/spaghettisoft/component/connectthefour/ConnectTheFour.java

```diff
@@ -0,0 +1,91 @@
+package com.spaghettisoft.component.connectthefour;
+
+import com.spaghettisoft.component.game.AbstractGame;
+import com.spaghettisoft.globals.StaticObjects;
+
+public class ConnectTheFour extends AbstractGame {
+
+	private Grid grid;
+	private boolean playerChange;
+
+	public ConnectTheFour() {
+		initialize();
+	}
+
+	@Override
+	protected void printEndGameMessage() {
+		drawGame();
+		if (playerChange) {
+			System.out.println("We have a winner! : Player " + 1);
+		} else {
+			System.out.println("We have a winner! : Player " + 2);
+		}
+		initialize();
+	}
+
+	@Override
+	protected void drawGame() {
+		for (int i = 0; i < grid.rowsLength(); i++) {
+			for (int j = 0; j < grid.columnsLength(); j++) {
+				System.out.print(grid.getElement(i, j) + " | ");
+			}
+			System.out.println();
+		}
+	}
+
+	@Override
+	protected boolean isEnded() {
+		boolean checkMessage = grid.check("X");
+		return checkMessage;
+	}
+
+	@Override
+	protected void processGame() {
+		int column = 0;
+		if (playerChange) {
+			do {

```

Comment:
@kr1994 
You can extract to function the code inside the "if". It will remove the duplication and will make the method more readable : )